static/app/views/detectors/components/forms/metricFormData.tsx

```diff
@@ -0,0 +1,311 @@
+// Note: Metric detectors only support MEDIUM and HIGH priority levels
+const PRIORITY_LEVEL_TO_DETECTOR_PRIORITY = {
+  [PriorityLevel.LOW]: 25, // DetectorPriorityLevel.LOW
+  [PriorityLevel.MEDIUM]: 50, // DetectorPriorityLevel.MEDIUM
+  [PriorityLevel.HIGH]: 75, // DetectorPriorityLevel.HIGH
+} as const;
+
+/**
+ * Creates escalation conditions based on priority level and available thresholds
+ */
+function createEscalationConditions(
+  data: MetricDetectorFormData
+): Array<Omit<DataCondition, 'id'>> {
+  const conditions: Array<Omit<DataCondition, 'id'>> = [];
+
+  if (!data.conditionType) {
+    return conditions;
+  }
+
+  // Always create the main condition for the initial priority level
+  if (data.conditionValue) {
+    // Convert LOW to MEDIUM since metric detectors don't support LOW
+    const effectiveInitialPriority =
+      data.initialPriorityLevel === PriorityLevel.LOW
+        ? PriorityLevel.MEDIUM
+        : data.initialPriorityLevel;
+
+    conditions.push({
+      type:
+        data.conditionType === 'gt' ? DataConditionType.GREATER : DataConditionType.LESS,
+      comparison: parseFloat(data.conditionValue) || 0,
+      conditionResult: PRIORITY_LEVEL_TO_DETECTOR_PRIORITY[effectiveInitialPriority],
+    });
+  }
+
+  // Create escalation conditions for higher priority levels
+  // Only add MEDIUM escalation if initial priority is LOW and mediumThreshold is provided
+  if (data.initialPriorityLevel === PriorityLevel.LOW && data.mediumThreshold) {
+    conditions.push({
+      type:
+        data.conditionType === 'gt' ? DataConditionType.GREATER : DataConditionType.LESS,
+      comparison: parseFloat(data.mediumThreshold) || 0,
+      conditionResult: PRIORITY_LEVEL_TO_DETECTOR_PRIORITY[PriorityLevel.MEDIUM],
+    });
+  }
+
+  // Only add HIGH escalation if initial priority is LOW or MEDIUM and highThreshold is provided
+  if (
+    (data.initialPriorityLevel === PriorityLevel.LOW ||
+      data.initialPriorityLevel === PriorityLevel.MEDIUM) &&
+    data.highThreshold
+  ) {
+    conditions.push({
+      type:
+        data.conditionType === 'gt' ? DataConditionType.GREATER : DataConditionType.LESS,
+      comparison: parseFloat(data.highThreshold) || 0,
+      conditionResult: PRIORITY_LEVEL_TO_DETECTOR_PRIORITY[PriorityLevel.HIGH],
+    });
+  }
+
+  return conditions;
+}
+
+/**
+ * Creates a resolution condition if resolveThreshold is provided
+ */
+function createResolutionCondition(
+  data: MetricDetectorFormData
+): Omit<DataCondition, 'id'> | null {
+  if (
+    data.resolveThreshold === undefined ||
+    data.resolveThreshold === '' ||
+    !data.conditionType
+  ) {
+    return null;
+  }
+
+  // Resolution condition uses opposite comparison type
+  // Backend only supports 'gt' (GREATER) and 'lt' (LESS), not 'lte' (LESS_OR_EQUAL)
+  const resolveConditionType =
+    data.conditionType === 'gt'
+      ? DataConditionType.LESS // Use LESS instead of LESS_OR_EQUAL
+      : DataConditionType.GREATER;
```

Comment:
i'm not sure if this logic doesn't seem quite right; if it's `gt` that is >, so the resolution would need to be <= (lte) -- so we'd need to account for the 4 types here rather than 2 (gt, gte, lt, lte)